tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -533,6 +627,67 @@ public:
       return newInterface;
    }
 
+   ////////////////////////////////////////////////////////////////////////////
+   /// \brief In case the value in the given column is missing, provide a default value
+   /// \tparam T The type of the column
+   /// \param[in] column Column name where missing values should be replaced by the given default value
+   /// \param[in] defaultValue Value to provide instead of a missing value
+   /// \return The node of the graph that will provide a default value
+   ///
+   /// This operation is useful in case an entry of the dataset is incomplete,
+   /// i.e. if one or more of the columns do not have valid values. It does not
+   /// modify the values of the column, but in case any entry is missing, it
+   /// will provide the default value to downstream nodes instead.
+   ///
+   /// Use cases include:
+   /// * When processing multiple files, one or more of them is missing a column
+   /// * In horizontal joining with entry matching, a certain dataset has no
+   ///   match for the current entry.
+   ///
+   /// ### Example usage:
+   ///
+   /// \code{.cpp}
+   /// // Assume a dataset with columns [idx, x] matching another dataset with
+   /// // columns [idx, y]. For idx == 42, the right-hand dataset has no match
+   /// ROOT::RDataFrame df{dataset};
+   /// auto df_default = df.DefaultValueFor("y", 33)
+   ///                     .Define("z", [](int x, int y) { return x + y; }, {"x", "y"});
+   /// auto colz = df_default.Take<int>("z");
+   /// \endcode
+   ///
+   /// \code{.py}
+   /// df = ROOT.RDataFrame(dataset)
+   /// df_default = df.DefaultValueFor("y", 33).Define("z", "x + y")
+   /// colz = df_default.Take[int]("z")
+   /// \endcode
+   template <typename T>
+   RInterface<Proxied, DS_t> DefaultValueFor(std::string_view column, const T &defaultValue)
+   {
+      RDFInternal::CheckForNoVariations("DefaultValueFor", column, fColRegister);
+
+      const auto validColumnNames = ColumnNames_t{column.data()};
+      CheckAndFillDSColumns(validColumnNames, TTraits::TypeList<T>{});
+
+      // Declare return type to the interpreter, for future use by jitted actions
+      auto retTypeName = RDFInternal::TypeID2TypeName(typeid(T));
+      if (retTypeName.empty()) {
+         // The type is not known to the interpreter.
+         // We must not error out here, but if/when this column is used in jitted code
+         const auto demangledType = RDFInternal::DemangleTypeIdName(typeid(T));
+         retTypeName = "CLING_UNKNOWN_TYPE_" + demangledType;
+      }
+
+      auto newColumn = std::make_shared<ROOT::Internal::RDF::RDefaultValueFor<T>>(
+         column, retTypeName, defaultValue, validColumnNames, fColRegister, *fLoopManager);
+
+      RDFInternal::RColumnRegister newCols(fColRegister);
+      newCols.AddDefine(std::move(newColumn));
```

Comment:
This seems to add a utility columns to implement the default.  How does they appear (or not) if one print (the list of) columns?

Reply:
The list of available column names is unchanged. The `DefaultValueFor` function takes in an input column name which must already exist in the dataset, otherwise it doesn't make sense. From the internal computation graph management point of view, the column reader connected to this column name is refreshed and will point to the one produced by DefaultValueFor. This pattern is inspired heavily from the existing implementation of Redefine

Reply:
But, thanks to your comment I now see that I'm not properly checking that the input column name already exists. I will add the extra check and a unit test, thanks!

Reply:
And of course there was a simple reason for that :) the column name might not exist at this point because e.g. it is missing from the first file of the chain, but we have to account for the fact it could be found in following files of the chain. So, the logic here must get a bit more sofisticated, investigating...

Reply:
My current belief is that it's actually a bad idea to check for the name of the column to already exist. For actual data sources like TTree or RNTuple, it does not make sense at all as I explained above. What's left is the case of a virtual dataset based on just entries in-memory. There we can potentially say that we know that a non-existing column name  at the beginning of the event loop cannot exist at any point of the event loop. But if we error out in that case we actually break the main point of `DefaultValueFor` which is to provide a default value for a given column name in case the input value is missing, no matter what. So I argue this should also be extended to in-memory only datasets. I will include it in the next commit.

Reply:
Or, actually, we could just decide that `DefaultValueFor` does not make sense at all if there is no input data source in the first place, that check is easy to do. What do you think @pcanal ?

Reply:
It sounds like in this case (no input data source) there would be no visible difference between `DefaultValueFor` and `Define` (i.e. the value would be used all the time).  So one hand it would be a duplication of features and the other hand having could be useful to have the same `RDataFrame` code work on both a mock-up and a real source.

Reply:
Thanks for your comment, this is actually a very interesting point. I would then frame it this way:
- In this PR, I disable the functionality for empty data sources by throwing an exception only for that case.
- In the future, it can be easily extended on user demand by making `DefaultValueFor` behave just like a `Define` with a single value in case of the empty data source and no existing column name.